src/components/Search.js

```diff
@@ -64,26 +68,35 @@ export const Search = () => {
     })
   }
 
+  const allowUsage = () => {
+    return isOnline || consentGranted
+  }
+
   useSoftkey('Search', {
-    right: i18n('softkey-settings'),
-    onKeyRight: () => { window.location.hash = '/settings' },
+    right: allowUsage() ? i18n('softkey-settings') : '',
+    onKeyRight: allowUsage() ? () => { window.location.hash = '/settings' } : null,
     center: current.type === 'DIV' ? i18n('centerkey-select') : '',
     onKeyCenter,
     onKeyLeft: isRandomEnabled() ? goToRandomArticle : null,
     onKeyBackspace: !(query && current.type === 'INPUT') && onExitConfirmDialog
-  }, [current.type, query])
+  }, [current.type, query, isOnline])
```

Comment:
I think we should merge this PR before #313, so that I can then resolve conflicts in the second PR which would clean this up a bit